cre.cpp

```diff
@@ -3495,14 +3495,6 @@ static int getImageDataFromPosition(lua_State *L) {
     return 0;
 }
 
-static int setUserHyphenationDict(lua_State *L) {
-    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
-    const char *filename = luaL_checkstring(L, 2);
-    int no_sloppy_load = luaL_checkint(L, 3);
-    UserHyphenDict::init(filename, no_sloppy_load !=0);
-    return 0;
-}
```

Comment:
Oh, you were really not using this function?
I somehow still prefer this one to the trick of hacking crengine.textlang.hyphenation.user.dict and appending -reload.
The support for setStringProperty("crengine.textlang.hyphenation.user.dict") in crengine is good to allow it to be used by other frontends (who would probably not implement editing) - but we're allowed to access lower level objects like UserHyphenDict in cre.cpp if we need to do a bit more that what we can do with setStringProperty.

Reply:
Yes I don't need that function. 
The cool thing on the string.prop is, that I can apply a rerender if the dict changes. (And we need the rerender, because some heights of paragraphs might change.

Reply:
Yes, I understand the need for rerender.
What I mean is there is nothing currently like this for string properties in crengine:
https://github.com/koreader/crengine/blob/362fe344fa01dec188f33606f0257006594ea316/crengine/src/lvdocview.cpp#L6682-L6689
These properties get a single argument, and it's somehow stored/saved in CoolReader settings db.

If you need a 2nd argument (reload=true/false), it feels ugly to hack this string property to acccept dynamic strings :)
It feels more proper to do it like you did in that bit of code you removed - so I would keep it and use it in credocument.lua instead of the string property. (But keep the string property handling in crengine lvdocvien.cpp, without the "-reload" hack, as a normal CoolReader setting, without the ability to force a reload.)

Reply:
What about one function to load the dict and one property just to rerender?

Reply:
Not sure what you mean by "rerender" (reloading the dict ? because re-rendering the book is never explicitely asked, it's done on page drawing by crengine when the hashes have changed).

But what's the issue with the above removed function? It seems just ideal as it was, no ?

Reply:
Now, when I call the setproperty the dict is  loaded the hash changes and I call explicitly RENDER_REQUEST

Reply:
Ah, it's not auto re-rendered when you just set a new updated dict, as there was yet no need to check hash as not re-drawing? Just curious: would it be re-rendered when you go to the next page (then, their must be a rendering hashes check).
Anyway, you could do in the above function what's done for the string property:
```c
if ( UserHyphenDict::init(filename, no_sloppy_load !=0) ) {
    doc->text_view->requestRender(); /* this is public, and what the macro REQUEST_RENDER does */
}
```

Reply:
Strange, it seems, that the rerendering works even without a `requestRender` now. Maybe that was necessary before I implemented the hash? Don't know anymore.